src/utils.jl

```diff
@@ -267,3 +266,166 @@ increment(n::Storage, t_inv) = (
 Returns the discount rate of `EnergyModel` modeltype
 """
 discount_rate(modeltype::AbstractInvestmentModel) = modeltype.r
+
+"""
+    has_investment(n::Storage, field::Symbol)
+
+For a given `Storage` node, checks that it contains investments for the field `field`, that
+is `:charge`, `:level`, or `:discharge`.
+"""
+function has_investment(n::Storage, field::Symbol)
+    (
+        hasproperty(n, :data) &&
+        !isempty(filter(data -> typeof(data) <: InvestmentData, n.data)) &&
+        !isnothing(getproperty(investment_data(n), field))
+    )
+end
+
+"""
+    investment_data(type, field::Symbol)
+
+Return the investment data of the type `type`.
+"""
+investment_data(type, field::Symbol) = getproperty(investment_data(type), field)
+
+"""
+    investment_mode(inv_data::GeneralInvData)
+
+Return the investment mode of the investment data `inv_data`. By default, all investments
+are continuous.
+"""
+investment_mode(inv_data::GeneralInvData) = inv_data.inv_mode
+
+"""
+    lifetime_mode(inv_data::GeneralInvData)
+
+Return the lifetime mode of the investment data `inv_data`. By default, all investments
+are unlimited.
+"""
+lifetime_mode(inv_data::GeneralInvData) = inv_data.life_mode
+
+"""
+    lifetime(inv_data::GeneralInvData)
+
+Return the lifetime of the investment data `inv_data` as `TimeProfile`.
+"""
+lifetime(inv_data::GeneralInvData) = inv_data.lifetime
+
+"""
+    lifetime(inv_data::GeneralInvData, t)
+
+Return the lifetime of the investment data `inv_data` in period `t`.
+"""
+lifetime(inv_data::GeneralInvData, t) = inv_data.lifetime[t]
+
+start_cap(n::EMB.Node, t_inv, inv_data::StartInvData, field, modeltype::EnergyModel) =
+    inv_data.initial
+start_cap(n::EMB.Node, t_inv, inv_data::NoStartInvData, field, modeltype::EnergyModel) =
+    capacity(n, t_inv)
+start_cap(n::Storage, t_inv, inv_data::StartInvData, field, modeltype::EnergyModel) =
```

Comment:
Is this identical to the EMB.Node one and is redundant?

Reply:
Yes, it is redundant, I will remove it and adjust the call for EMB.Node..